.pre-commit-config.yaml

```diff
@@ -20,7 +20,15 @@ repos:
     rev: 7.0.0
     hooks:
       - id: flake8
-  - repo: https://github.com/pre-commit/mirrors-eslint
-    rev: v9.2.0
+  - repo: local
```

Comment:
@adamchainz can you explain how this resolves the issue? 
The way I understand the issue is that you now need to run `npm install` for the commit hook to work as this installs `globals` as a dev dependency. With this change, if I run `npm uninstall globals` and run this, I get the same errors

One way forward we could consider would be to remove our own dependency to the globals package and defining our globals in [`eslint.config.js`](https://eslint.org/docs/latest/use/configure/language-options#using-configuration-files)

Reply:
`pre-commit run eslint -a` passes on `main` for me

Reply:
Ah, you know what, I was mistaken. I was acting based on my experience of one of the ESLint beta or RC releases requiring this change, rather than testing here. Yes, it works on `main`, because ESLint depends on both `@eslint/js` and `globals`: https://github.com/eslint/eslint/blob/main/package.json . Sorry for the noise.